app/model/reports/QualifiedPercentagesReport.scala

```diff
@@ -0,0 +1,131 @@
+package model.reports
+
+import ai.x.play.json.Jsonx
+import model.Campaign
+import org.joda.time.DateTime
+import play.api.Logger
+import play.api.libs.json.Format
+import repositories._
+
+import scala.concurrent.Future
+
+case class QualifiedMetricReport(total: Long, qualifiedCount: Long, percentage: Double)
+
+object QualifiedMetricReport {
+  implicit val qualifiedMetricReportFormat: Format[QualifiedMetricReport] = Jsonx.formatCaseClass[QualifiedMetricReport]
+}
+
+case class QualifiedPercentagesReport(campaignId: String, metrics: Map[String, QualifiedMetricReport]) {
+
+  def refresh = QualifiedPercentagesReport.generateReport(campaignId)
+
+}
+
+object QualifiedPercentagesReport {
+
+  implicit val ec = AnalyticsDataCache.analyticsExectuionContext
+
+  implicit val qualifiedPercentagesReportFormat: Format[QualifiedPercentagesReport] = Jsonx.formatCaseClass[QualifiedPercentagesReport]
+
+  def getQualifiedPercentagesReportForCampaign(campaignId: String): Option[QualifiedPercentagesReport] = {
+
+    AnalyticsDataCache.getCampaignQualifiedPercentagesReport(campaignId) match {
+      case Hit(report) => {
+        Logger.debug(s"getting qualified percentages for campaign $campaignId - cache hit")
+        Some(report)
+      }
+      case Stale(report) => {
+        Logger.debug(s"getting qualified percentages for campaign $campaignId - cache stale spawning async refresh")
+
+        Future{
+          Logger.debug(s"async refresh of qualified percentages for campaign $campaignId")
+          report.refresh
+        } // serve stale but spawn refresh future
+        Some(report)
+      }
+      case Miss => {
+        Logger.debug(s"getting qualified percentages for campaign $campaignId - cache miss fetching sync")
+
+        generateReport(campaignId)
+      }
+    }
```

Comment:
I wonder if we can reuse this caching pattern match as it seems to recur with slight variations?  But the variations may make it not worthwhile.